testing/integration/otel_helm_test.go

```diff
@@ -277,3 +194,90 @@ func k8sStepCheckRunningPods(podLabelSelector string, expectedPodNumber int, con
 		}, 5*time.Minute, 10*time.Second, fmt.Sprintf("at least %d agent containers should be checked", expectedPodNumber))
 	}
 }
+
+func k8sStepDeployJavaApp(info *define.Info, dsType, dataset, datastreamNamespace string) k8sTestStep {
```

Comment:
Since this does 2 different things, deploying the target object and then checks the ES data-stream, should that be renamed or better split in 2 separate ones? I guess `k8sStepCheckDatastreamsHits` could be reused for the second part right? So I would just keep the deployment of the target in this one and then call from the main flow the  `k8sStepCheckDatastreamsHits`. WDYT?

Reply:
good point, wrapped both steps in a single main function https://github.com/elastic/elastic-agent/pull/7754/commits/788eef8c4a36b3ae2f8184c7d50ca0e6d75e7467